components/Sidebar.go

```diff
@@ -0,0 +1,230 @@
+}
+
+func NewSidebar() *Sidebar {
+	flex := tview.NewFlex().SetDirection(tview.FlexColumnCSS)
+	frame := tview.NewFrame(flex)
+	frame.SetBackgroundColor(tview.Styles.PrimitiveBackgroundColor)
+	frame.SetBorder(true)
+
+	sidebarState := &SidebarState{
+		currentFieldIndex: 0,
+	}
+
+	newSidebar := &Sidebar{
+		Frame:       frame,
+		Flex:        flex,
+		state:       sidebarState,
+		Fields:      []*tview.TextArea{},
+		subscribers: []chan models.StateChange{},
+	}
+
+	newSidebar.SetInputCapture(newSidebar.inputCapture)
+
+	newSidebar.SetBlurFunc(func() {
+		newSidebar.SetCurrentFieldIndex(0)
+	})
+
+	return newSidebar
+}
+
+func (sidebar *Sidebar) AddField(title, text string, fieldWidth int) {
+	field := tview.NewTextArea()
+	field.SetWrap(true)
+	field.SetDisabled(true)
+
+	field.SetBorder(true)
+	field.SetTitle(title)
+	field.SetTitleAlign(tview.AlignLeft)
+	field.SetTitleColor(tview.Styles.PrimaryTextColor)
+	field.SetText(text, true)
+	field.SetTextStyle(tcell.StyleDefault.Background(tview.Styles.PrimitiveBackgroundColor).Foreground(tview.Styles.SecondaryTextColor))
+
+	textLength := len(field.GetText())
+
+	itemFixedSize := 3
+
+	if textLength >= fieldWidth*3 {
+		itemFixedSize = 5
+	} else if textLength >= fieldWidth {
+		itemFixedSize = 4
+	} else {
+		field.SetWrap(false)
+	}
+
+	sidebar.Fields = append(sidebar.Fields, field)
+	sidebar.Flex.AddItem(field, itemFixedSize, 0, true)
+}
+
+func (sidebar *Sidebar) FocusNextField() {
+	newIndex := sidebar.GetCurrentFieldIndex() + 1
+
+	if newIndex < sidebar.Flex.GetItemCount() {
+		item := sidebar.Fields[newIndex]
+
+		if item != nil {
+			sidebar.SetCurrentFieldIndex(newIndex)
+			App.SetFocus(item)
+			App.ForceDraw()
+			return
+		}
+
+	}
+}
+
+func (sidebar *Sidebar) FocusPreviousField() {
+	newIndex := sidebar.GetCurrentFieldIndex() - 1
+
+	if newIndex >= 0 {
+		item := sidebar.Fields[newIndex]
+
+		if item != nil {
+			sidebar.SetCurrentFieldIndex(newIndex)
+			App.SetFocus(item)
+			App.ForceDraw()
+			return
+		}
+	}
+}
+
+func (sidebar *Sidebar) FocusFirstField() {
+	sidebar.SetCurrentFieldIndex(0)
+	App.SetFocus(sidebar.Fields[0])
+}
+
+func (sidebar *Sidebar) FocusLastField() {
+	newIndex := sidebar.Flex.GetItemCount() - 1
+	sidebar.SetCurrentFieldIndex(newIndex)
+	App.SetFocus(sidebar.Fields[newIndex])
+}
+
+func (sidebar *Sidebar) FocusField(index int) {
+	sidebar.SetCurrentFieldIndex(index)
+	App.SetFocus(sidebar.Fields[index])
+}
+
+func (sidebar *Sidebar) Clear() {
+	sidebar.Fields = make([]*tview.TextArea, 0)
+	sidebar.Flex.Clear()
+}
+
+func (sidebar *Sidebar) EditTextCurrentField() {
+	index := sidebar.GetCurrentFieldIndex()
+	item := sidebar.Fields[index]
+
+	sidebar.SetEditingStyles(item)
+}
+
+func (sidebar *Sidebar) inputCapture(event *tcell.EventKey) *tcell.EventKey {
+	command := app.Keymaps.Group(app.SidebarGroup).Resolve(event)
+
+	switch command {
+	case commands.UnfocusSidebar:
+		sidebar.Publish(models.StateChange{Key: "Unfocusing", Value: nil})
```

Comment:
Here is an example of where you should reuse the constants I was talking about

https://github.com/jorgerojas26/lazysql/pull/103#discussion_r1749229328